specification/cognitiveservices/data-plane/Personalizer/preview/v1.1-preview.2/Personalizer.json

```diff
@@ -0,0 +1,1895 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "v1.1-preview.2",
+    "title": "Personalizer Base",
+    "description": "Personalizer Service is an Azure Cognitive Service that makes it easy to target content and experiences without complex pre-analysis or cleanup of past data. Given a context and featurized content, the Personalizer Service returns which content item to show to users in rewardActionId. As rewards are sent in response to the use of rewardActionId, the reinforcement learning algorithm will improve the model and improve performance of future rank calls."
+  },
+  "basePath": "/personalizer/v1.1-preview.2",
+  "schemes": [
+    "https"
+  ],
+  "paths": {
+    "/configurations/service": {
+      "get": {
+        "tags": [
+          "ConfigurationsV1Dot1Preview1"
+        ],
+        "summary": "Get Service Configuration.",
+        "description": "Get the Personalizer service configuration.",
+        "operationId": "ServiceConfiguration_Get",
+        "consumes": [],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ServiceConfiguration"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Successful ServiceConfiguration_Get request": {
+            "$ref": "./examples/ServiceConfiguration_Get.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "ConfigurationsV1Dot1Preview1"
+        ],
+        "summary": "Update Service Configuration.",
+        "description": "Update the Personalizer service configuration.",
+        "operationId": "ServiceConfiguration_Update",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "config",
+            "in": "body",
+            "description": "The personalizer service configuration.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ServiceConfiguration"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ServiceConfiguration"
+            }
+          },
+          "default": {
+            "description": "Updating defaultReward, rewardWaitTime and rewardAggregation when changing learning mode from Online to Apprentice mode and vice versa is not allowed. Make the mode change and then change the additional settings with an additional API call.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
```

Comment:
@slahabar -- Please review the [Handling Errors](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#Handling-Errors) section of the guidelines. Error responses should return the ```x-ms-error-code``` header. 

Note: Please reference the [Breaking Change Guidelines](https://microsoft.sharepoint.com/teams/azureecosystem/servicerings/Shared%20Documents/Breaking%20Changes/Azure%20Breaking%20Changes%20Program%20Definition-%20V1.2.pdf) as well if you add the header to existing operations.

Reply:
@slahabar -- here's an example of using the ```x-ms-error-code header```, it's from [blob storage](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/storage/data-plane/Microsoft.BlobStorage/preview/2020-10-02/blob.json). 
```
 "default": {
            "description": "Failure",
            "headers": {
              "x-ms-error-code": {
                "x-ms-client-name": "ErrorCode",
                "type": "string"
              }
            },
            "schema": {
              "$ref": "#/definitions/StorageError"
            }
          }
        }
```

It's a best practice to add an enum with your known errors. However,  you can leave them as strings b/c it's easier for downstream code gen.

Reply:
Personalizer will create a separate task to address this issue.
Link task here